realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -385,12 +392,18 @@ private boolean categorizeField(Element element) {
             if (!categorizePrimaryKeyField(field)) { return false; }
         }
 
-        // Check @LinkingObjects last since it is not allowed to be either @Index, @Required or @PrimaryKey
+        // @LinkingObjects cannot be @PrimaryKey or @Index.
         if (field.getAnnotation(LinkingObjects.class) != null) {
+            // Do not add backlinks to fields list.
             return categorizeBacklinkField(field);
         }
 
-        // Standard field that appear valid (more fine grained checks might fail later).
+        // Similarly, a RealmInteger cannot be a @PrimaryKey, @Index or @LinkingObject.
```

Comment:
I'm not sure it is check completely, yet.  It will be though.  Perhaps a *FIXME* is in order...

As far as indexing, I don't actually see a reason to forbid it.  Seems odd to me but I can certainly allow it!